src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs

```diff
@@ -205,15 +205,15 @@ public void AttachAndModifyCompanyApplications(IEnumerable<(Guid companyApplicat
             .SingleOrDefaultAsync();
 
     /// <inheritdoc />
-    public Task<(Guid CompanyId, string CompanyName, string? BusinessPartnerNumber, IEnumerable<string> IamIdpAliasse, CompanyApplicationTypeId ApplicationTypeId, Guid? NetworkRegistrationProcessId)> GetCompanyAndApplicationDetailsForApprovalAsync(Guid applicationId) =>
+    public Task<(Guid CompanyId, string CompanyName, string? BusinessPartnerNumber, IEnumerable<IdentityProviderForTheme> IamIdps, CompanyApplicationTypeId ApplicationTypeId, Guid? NetworkRegistrationProcessId)> GetCompanyAndApplicationDetailsForApprovalAsync(Guid applicationId) =>
         portalDbContext.CompanyApplications.Where(companyApplication =>
                 companyApplication.Id == applicationId &&
                 companyApplication.ApplicationStatusId == CompanyApplicationStatusId.SUBMITTED)
-            .Select(ca => new ValueTuple<Guid, string, string?, IEnumerable<string>, CompanyApplicationTypeId, Guid?>(
+            .Select(ca => new ValueTuple<Guid, string, string?, IEnumerable<IdentityProviderForTheme>, CompanyApplicationTypeId, Guid?>(
                 ca.CompanyId,
                 ca.Company!.Name,
                 ca.Company.BusinessPartnerNumber,
-                ca.Company.IdentityProviders.Select(x => x.IamIdentityProvider!.IamIdpAlias),
+                ca.Company.IdentityProviders.Select(x => new IdentityProviderForTheme(x.IamIdentityProvider!.IamIdpAlias, x.IamIdentityProvider!.IdentityProvider!.IdentityProviderTypeId)),
```

Comment:
as the aliases are only used to set the valid shared idp themes the selection of aliases should be restricted to precisely those:
```suggestion
                ca.Company.IdentityProviders.Where(x => x.IdentityProviderTypeId == IdentityProviderTypeId.SHARED && x.IamIdentityProvider != null).Select(x => x.IamIdentityProvider!.IamIdpAlias),

```